pkg/controller/volume/attachdetach/cache/desired_state_of_world.go

```diff
@@ -313,6 +324,30 @@ func (dsw *desiredStateOfWorld) VolumeExists(
 	return false
 }
 
+// DetermineVolumeAction determines action to be taken for pod volume
+// if the volume should be added or removed from desired state of world.
+// return values:
+//    true  -- if volume needs to be attached
+//    false -- if volume needs to be detached
+func (dsw *desiredStateOfWorld) DetermineVolumeAction(pod *v1.Pod, defaultValue bool) bool {
```

Comment:
This is a really funky method to add to `desired_state_of_the_world.go`--it blurs the boundary of what this class is responsible for. Think about what the responsibilities of this code is and if it makes sense for this logic to live here, and use that to refactor this. If you want my thoughts on how this should be factored let me know.